### PR TITLE
Add Jenkins CI with Cuda and HIP

### PR DESCRIPTION
**Pull request by streeve**

### CODE REVIEW

**Comment by streeve** [comment]
test this please.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    KEYDUMP_FILE=keydump && \
+    wget --quiet ${KEYDUMP_URL}/${KEYDUMP_FILE} && \
+    wget --quiet ${KEYDUMP_URL}/${KEYDUMP_FILE}.sig && \
+    gpg --import ${KEYDUMP_FILE} && \
+    gpg --verify ${KEYDUMP_FILE}.sig ${KEYDUMP_FILE} && \
+    rm ${KEYDUMP_FILE}*
+
+# Install CMake
+ENV CMAKE_DIR=/opt/cmake
+RUN CMAKE_VERSION=3.13.4 && \
```
Comment (dalg24) [inline docker/Dockerfile:26]: I would go for something more recent.  Kokkos 3.4 will require 3.16

Comment (streeve) [reply docker/Dockerfile]: Good point, went to 3.18

**Comment by streeve** [comment]
Ok @sslattery with Damien's help everything is passing except `LevelSetRedistance_test_CUDA_1` - is that allocation error obvious to you?

**Comment by streeve** [comment]
ping @sslattery still the one Cuda error for the vector scalar overload

@dalg24 can I force Jenkins to start the HIP build from scratch? I'm not sure what the issue is there

**Comment by dalg24** [comment]
Did you try to build the image locally?

**Comment by streeve** [comment]
> Did you try to build the image locally?

Yes, I was just able to build (at least the dependencies) on my mac.

**Comment by streeve** [comment]
@sslattery this passes because I disabled everything that was failing. I think we should get this merged and fix later

**Comment by streeve** [comment]
@abisner @kwitaechong @sslattery this is ready for review (and blocks everything else)

**Review by sslattery — APPROVED** [review]

**Pull request merged**